src/Transpile.ts

```diff
@@ -104,7 +104,8 @@ export function transpile({
             const isDeclaration = fileName.endsWith(".d.ts");
             const isDeclarationMap = fileName.endsWith(".d.ts.map");
             if (isDeclaration) {
-                updateTranspiledFile(sourceFile.fileName, { declaration: data });
+                const dataWithoutEmptyStatements = data.replace(/^;/gm, "");
```

Comment:
This looks quite hacky and likely breaks source maps (which is the main reason to use transformer anyway). Have you tried `ts.createNotEmittedStatement`? It requires original node reference, but `sourceFile` or `undefined!` should work. I've also tried to attach it to first first statement, but it doesn't look good when it already has some comments (though probably works)

Reply:
Brilliant! Didn't know about `NotEmittedStatement`. No semicolon is emitted so this change has been reverted